File: Java/InheritanceLowLevel_Java/SalesPerson.java
package com.epam.test.automation.java.practice6;
import java.math.BigDecimal;
/**
 * <summary>
 * Implement code according to description of task.
 * </summary>
 */
public class SalesPerson extends Employee {

    private int percent;
    public SalesPerson(String name, BigDecimal salary, int percent) {
        super(name, salary);
        this.percent = percent;
    }
    
    @Override
    public void setBonus(BigDecimal bonus) {
        BigDecimal three =  new BigDecimal("3");
        BigDecimal two =  new BigDecimal("2");
        
        if (percent > 200){
            super.setBonus(bonus.multiply(three));
        } else if ( percent > 100 && percent <=200){
            super.setBonus(bonus.multiply(two));
        } else {
            super.setBonus(bonus);
        }
    
    }
}

File: Inheritance/SalesPerson.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace InheritanceTask
{
    class SalesPerson : Employee
    {
        private readonly int percent;
        public SalesPerson(string name, decimal salary, int percent) : base(name, salary)
        {
            this.percent = percent;
        }
        public override void SetBonus(decimal bonus)
        {
            if (percent > 100)
            {
                bonus *= 2;
            }
            if (percent > 200)
            {
                _ = bonus * 3;
            }
        }
    }
}
